internal/controllers/topology/cluster/patches/variables/variables_test.go

```diff
@@ -853,6 +854,373 @@ func TestMachineDeployment(t *testing.T) {
 	}
 }
 
+func TestMachinePool(t *testing.T) {
+	tests := []struct {
+		name                            string
+		mpTopology                      *clusterv1.MachinePoolTopology
+		forPatch                        string
+		variableDefinitionsForPatch     map[string]bool
+		mp                              *expv1.MachinePool
+		mpBootstrapTemplate             *unstructured.Unstructured
```

Comment:
Nit: we pass it later on as bootstrapObject to the `MachinePool` function.

Should we rename it here too?

```suggestion
		mpBootstrapObject             *unstructured.Unstructured
```

Reply:
I would suggest to even drop the Object suffix in accordance with: https://github.com/kubernetes-sigs/cluster-api/pull/9346#discussion_r1314693395. WDYT?

Reply:
Let's go with mpBootstrapConfig

I consider BootstrapConfig the correct one